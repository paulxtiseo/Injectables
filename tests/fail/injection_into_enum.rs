use injectables::{injectable, inject_fields};

#[injectable]
pub struct Base {
    pub id: u64,
}

#[inject_fields(Base)]
pub enum InvalidTarget {  // Should fail because enums can't receive injected fields
    Variant1,
    Variant2,
}

fn main() {}