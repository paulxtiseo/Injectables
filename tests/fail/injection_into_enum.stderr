error: Only structs are supported as injection targets
 --> tests/fail/injection_into_enum.rs:8:1
  |
8 | #[inject_fields(Base)]
  | ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `inject_fields` (in Nightly builds, run with -Z macro-backtrace for more info)
